### ISSUE TITLE
Cursor positioning on cycle tasks, Marking already closed tasks from menu

### ISSUE THREAD

**Issue by RobColes** [wenlzhang/obsidian-task-marker#10]
Apologies if this is a bit premature... I am already loving the plugin!  Couple of comments

1) If I cycle through the statuses with the cursor past the last of the text, the cursor position is left alone. If I am e.g. in the middle of a word, as soon as I cycle via hotkey, the cursor goes to the start of the line and stays there.  I would prefer the cursor position to be left alone if possible.

2) If the task is completed or cancelled, I can't then change my mind and mark it as anything else from the context menu - I have to mark the task back to open/blank first, and them mark to something else. Is it possible to not restrict this, or make it an option? For my use case it's an unnecessary extra step. I can toggle through cancelled and other statuses with the hotkey fine. 

Thanks
Rob

**Comment by wenlzhang** [comment]
Thank you for the comments!

> 1. If I cycle through the statuses with the cursor past the last of the text, the cursor position is left alone. If I am e.g. in the middle of a word, as soon as I cycle via hotkey, the cursor goes to the start of the line and stays there.  I would prefer the cursor position to be left alone if possible.

I need to investigate if I can resolve this issue. Otherwise, I am open to a pull request on this.

> If the task is completed or cancelled, I can't then change my mind and mark it as anything else from the context menu - I have to mark the task back to open/blank first, and them mark to something else. Is it possible to not restrict this, or make it an option? For my use case it's an unnecessary extra step. I can toggle through cancelled and other statuses with the hotkey fine.

If a task is completed/cancelled (assume that text would be appended), then is marked with another status specified in row 1 (text would be appended). This means that text would be appended twice to achieve the desired status. Is this the desired behaviour? Or, do you mean changing your mind immediately or maybe after some days?

**Comment by ebullient** [comment]
Can we merge this back with the task collector plugin?

The task state cycle comes from that plugin, where a "complete" state is the end of the cycle (somewhat final). It is deliberately hard to accidentally say something isn't complete anymore, but you can cycle through all of the other states at will.

It is unfortunate (to me) that this was forked, as we should have been able to find a way to add missing function with a few conversations or PRs, so I'm wondering where the functionality gap is (I believe it is the task cycle and significant doc revisions, not sure if there is anything else)

**Comment by wenlzhang** [comment]
> Can we merge this back with the task collector plugin?
> 
> The task state cycle comes from that plugin, where a "complete" state is the end of the cycle (somewhat final).

The core code for achieving this feature for cycling task statuses can be found as follows:
https://github.com/wenlzhang/obsidian-task-marker/blob/be4de99c624cf5ab341476073e18002020be6b9d/src/taskmarker-TaskMarker.ts#L531

Of course, I can make a pull request regarding this feature if needed. :-)

> It is unfortunate (to me) that this was forked, as we should have been able to find a way to add missing function with a few conversations or PRs, so I'm wondering where the functionality gap is (I believe it is the task cycle and significant doc revisions, not sure if there is anything else)

To be honest, I have been using Task Collector for quite a long time and have made some feature requests. Also, Task Marker was developed based on the Task Collector's code (really appreciate this). Nevertheless, the main reasons for me to develop this plugin are as follows:

- My personal workflow requires that marking tasks would also append text (similar to the case when completing/cancelling tasks), and I need this feature almost every day. For this, I made a feature request about 10 months ago, and I am not sure when this feature would be supported. Therefore, I added the feature of appending text when marking tasks. One can also choose not to append text for statuses assigned in the second row.
    - My feature request: https://github.com/ebullient/obsidian-task-collector/issues/36
    - A similar request: https://github.com/ebullient/obsidian-task-collector/issues/56
- Similarly, I would like to mark tasks as `[>]` with hotkeys. I guess this feature is quite personal. Therefore, I added the feature of marking individual task statuses with hotkeys.
- I was eager to learn how to develop Obsidian plugins so that (hopefully) my needs on task management and other things can be satisfied. Again, this is a quite personal matter.

As for the differences between Task Marker and Task Collector, there are a few of them as of now:

- One can use hotkeys and context menu to create tasks, i.e. convert a list item as an open task, and append text.
    - This is because I would like to know when a task is created.
- One can append text when marking tasks. One can also use hotkeys to quickly change individual task statuses.
    - This is explained earlier.
- One can use hotkeys and context menu to cycle among task statuses.
    - This can be used for the case like changing to a desired status without choosing the specific task mark.
- One can append different types of text when creating, completing and marking tasks.
- One can append text to any line.
    - This is not specifically for task management, but as a way to quickly use hotkeys to append time stamp so that one can know when a certain line is written.
- Features like moving tasks and operating on all tasks are not kept in Task Marker.
    - This is because one can rely on Task Collector to achieve them.

Still, we can discuss how to merge certain features of Task Marker into Task Collector in the future if needed. Or, certain features can be kept separately in Task Marker, as most of them are quite personal needs, and I am not certain if they fit Task Collector's plan.

**Comment by ebullient** [comment]
Breaking up my original delineation between complete and incomplete tasks is something I've been thinking about for a long time. Some of the behavior (append/remove, reset, and state-machine semantics) were all anchored around that delineation in my mind, and I think it causes more confusion than it is worth.

I'm in the process of breaking that apart now, which would allow: 

- setting specific append text for one or more marks
- setting different completion areas for different marks (an open request)

With those settings separated, allowing separate command registration would be pretty easy. I guess I question individual context menus (because those menus can get long already). If you have the one selection modal that still accommodates different behaviors (and the possibility of a hotkey), is that enough? I guess the context menu is more important for reading/live-preview.. 

By adding text/timestamp anywhere, you mean something beyond what natural language plugin can do? There are a few plugins already that can do quick text expansion, but I guess it is the consistent task/date orientation that you're thinking of?

**Comment by wenlzhang** [comment]
> setting specific append text for one or more marks

I was also thinking about something like this before. However, now I am thinking about this from another aspect: would it be very informative to simply append different texts for different marks? More specifically, one can see from the appended texts that the status of a task is changed several times, but would that provide very useful information/context other than how many times the status is changed?

Personally, sometimes I would do something like this when changing the due date of a task. I would add something like the following in a sub-item of the corresponding task: This due date is changed because xxx. Also, I might link to the corresponding block/note, which can be some short note or a task. This provides some context that helps one know why the due date is changed. Similarly, if I were to change the status of a task, (when necessary) I might add text like this helping myself remember why I changed the status of the task. This might be more informative than just appending the text.

> I guess I question individual context menus (because those menus can get long already).

I agree with this. At least, it may not be necessary to add individual context menus for all marks.

> If you have the one selection modal that still accommodates different behaviors (and the possibility of a hotkey), is that enough?

I think that one selection modal is enough for marks that are not used that frequently. However, for certain marks like `[>]`, I use it very frequently (similar to `[x]` and `[-]`). The reason for using this status is that I do not need to care about the task (**Task A**) any more in its original location. Instead, I may link it to another task (**Task B**) as a sub-task (**Task B-1**) in the form of a block link. There are a few reasons for this:

- I would rather not modify the structure of the original tasks. Therefore, I would usually not copy and paste the task to another location.
- In the new location, **Task B** may consist of several sub-tasks, including **Task B-1**, **Task B-2**, etc.
- **Task B** may be in another note or the same note. **Task B** can be seen as an "entry note" for tasks.

Anyway, for such a frequent use case, I would like to be able to mark tasks as `[>]` with just one step, e.g. via hotkeys, just like completing/cancelling tasks. Using the selection modal would need two steps. I might be a bit two picky on this. 😂

> I guess it is the consistent task/date orientation that you're thinking of?

Yes, kind of. I am considering adding texts like `📝 2023-01-08T12:00`, `❎ 2023-01-08T12:00` and `✅ 2023-01-08T12:00`, which correspond to new, marked and completed text, respectively. For other plugins/tools, I am uncertain if they can add the additional emojis before the time stamp, or if one can use hotkeys to append such texts (again, if one appends such texts very frequently, this is necessary for me). Before developing this plugin, I use the text expander espanso for such a purpose, but now I use this plugin more.

The reason for adding text like `📝 2023-01-08T12:00` to a line of text, versus tasks, is that I would like to know when a specific line of text is added. In the context of the slip-box method, it may be encouraged to create a new note if one were to modify the content of a note. However, I find it too difficult to create and manage so many (atom) notes. An example note illustrating this is as follows. Similarly, adding text like `✅ 2023-01-08T12:00` would mean that this specific line of text is not of interest any more or is completed/archived.

Example note:
```
This line of text is added when the note is created. Therefore, no text is appended at the end of this line.

This line of text is added at a later time. Therefore, text is appended at the end of this line. 📝 2023-01-08T12:00

This line of text is not of interest any more. ✅ 2023-01-08T12:00
```

**Comment by wenlzhang** [comment]
Appending texts to any line of text is also related to the following request made to Dataview: https://github.com/blacksmithgu/obsidian-dataview/discussions/685. With this feature being supported in the future, one can query lines/blocks of notes and see if they contain desired information, e.g. keywords, completed or note, etc.

According to the description, Datacore, a successor to Dataview, might support querying lines/blocks of notes.
https://github.com/blacksmithgu/datacore

**Comment by ebullient** [comment]
> > setting specific append text for one or more marks
> 
> I was also thinking about something like this before. However, now I am thinking about this from another aspect: would it be very informative to simply append different texts for different marks? More specifically, one can see from the appended texts that the status of a task is changed several times, but would that provide very useful information/context other than how many times the status is changed?

I could make a simple toggle for whether or not to "undo" the previous append. If I didn't, it would keep appending. 

> 
> Personally, sometimes I would do something like this when changing the due date of a task. I would add something like the following in a sub-item of the corresponding task: This due date is changed because xxx. Also, I might link to the corresponding block/note, which can be some short note or a task. This provides some context that helps one know why the due date is changed. Similarly, if I were to change the status of a task, (when necessary) I might add text like this helping myself remember why I changed the status of the task. This might be more informative than just appending the text.

I think trying to put all of this in the purview of a hot-key press is not actually saving that much time. How much context is (or is not) important will change as the nature of the task changes. For example, if I did add a toggle of some kind (for all task types? for just one?) to stop removing previously appended text.. at some point none of that information is useful anymore (or it isn't presented in a usable or readable way)...

> 
> > I guess I question individual context menus (because those menus can get long already).
> 
> I agree with this. At least, it may not be necessary to add individual context menus for all marks.
> 
> > If you have the one selection modal that still accommodates different behaviors (and the possibility of a hotkey), is that enough?
> 
> I think that one selection modal is enough for marks that are not used that frequently. However, for certain marks like `[>]`, I use it very frequently (similar to `[x]` and `[-]`). The reason for using this status is that I do not need to care about the task (**Task A**) any more in its original location. Instead, I may link it to another task (**Task B**) as a sub-task (**Task B-1**) in the form of a block link. There are a few reasons for this:
> 
>     * I would rather not modify the structure of the original tasks. Therefore, I would usually not copy and paste the task to another location.
> 
>     * In the new location, **Task B** may consist of several sub-tasks, including **Task B-1**, **Task B-2**, etc.
> 
>     * **Task B** may be in another note or the same note. **Task B** can be seen as an "entry note" for tasks.
> 
> 
> Anyway, for such a frequent use case, I would like to be able to mark tasks as `[>]` with just one step, e.g. via hotkeys, just like completing/cancelling tasks. Using the selection modal would need two steps. I might be a bit two picky on this. 😂

Hotkeys (and context menu additions) are in the mix for my revisions, so .. we'll see what you think. ;)

> 
> > I guess it is the consistent task/date orientation that you're thinking of?
> 
> Yes, kind of. I am considering adding texts like `📝 2023-01-08T12:00`, `❎ 2023-01-08T12:00` and `✅ 2023-01-08T12:00`, which correspond to new, marked and completed text, respectively. For other plugins/tools, I am uncertain if they can add the additional emojis before the time stamp, or if one can use hotkeys to append such texts (again, if one appends such texts very frequently, this is necessary for me). Before developing this plugin, I use the text expander espanso for such a purpose, but now I use this plugin more.
> 
> The reason for adding text like `📝 2023-01-08T12:00` to a line of text, versus tasks, is that I would like to know when a specific line of text is added. In the context of the slip-box method, it may be encouraged to create a new note if one were to modify the content of a note. However, I find it too difficult to create and manage so many (atom) notes. An example note illustrating this is as follows. Similarly, adding text like `✅ 2023-01-08T12:00` would mean that this specific line of text is not of interest any more or is completed/archived.
> 
> Example note:
> 
> ```
> This line of text is added when the note is created. Therefore, no text is appended at the end of this line.
> 
> This line of text is added at a later time. Therefore, text is appended at the end of this line. 📝 2023-01-08T12:00
> 
> This line of text is not of interest any more. ✅ 2023-01-08T12:00
> ```

This is an interesting use case. For commands (hot-keyable) in edit mode, a flag could allow text to be appended to a line that isn't a task (turn off any "find the task" matching capability and just append.. ?). Do the strings that you use in this case also overlap with task types?

**Comment by RobColes** [comment]
>If a task is completed/cancelled (assume that text would be appended), then is marked with another status specified in row 1 (text would be appended). This means that text would be appended twice to achieve the desired status. Is this the desired behaviour? Or, do you mean changing your mind immediately or maybe after some days?

My use case doesn't automatically append anything, I am really just looking at changing the status so multiple appends aren't an issue for me. 

I do find it frustrating when this sort of "nannying" is done. This may just be me, but if I've right clicked to pull up a dialog and then selected a value, it's because I actually want that value and I'm OK with the consequences. I don't see the benefit in making it "somewhat hard" if the task has been closed. Maybe this behaviour could be toggleable? or an optional "are you sure?" Or worst case disable the other statuses so at least it is obvious that the selection isn't available rather than it being silently ignored. Apologies for the rant :-)

Most of my use of the different statuses is within a daily note, so it may be that I think I've finished a ticket, but the user comes back with a further issue, or occasionally a cancelled call gets re-instated. Changes are within the day rather than longer term. I don't tend to go back to the daily note to update dates, and I normally manage dates on other notes with text expander macros. 

The hot key status toggle is the killer feature for me. I wasn't aware of task collector. 

I am in awe of the generosity of authors sharing their code so that it can be tweaked and extended. Thank you both!

**Comment by ebullient** [comment]
> This may just be me, but if I've right clicked to pull up a dialog and then selected a value, it's because I actually want that value and I'm OK with the consequences.

I'm in the process of removing it. It has roots in how/why I created the plugin in the first place, but I actually agree with you now (it annoys me, too). ;) I'll comment here when there is a BRAT release ready, and you can give it a spin.

**Comment by RobColes** [comment]
Thank you!

**Comment by ebullient** [comment]
Would either of you be willing to send me your plugin config (data.json)? A temporary gist would work: https://docs.github.com/en/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists (or cut/paste here) .. just want some examples besides my own, just in case. ;)

**Comment by RobColes** [comment]
Not sure if this is what you mean, and it's the config for task marker, but this is what I have:
https://gist.github.com/RobColes/a35b3c4a8fcb150aa33d185ce1ae40d8

**Comment by ebullient** [comment]
Not in BRAT yet, but.... 

Please use this in a test vault! The settings are much different (and there is a lot more I can do next), but please give this a spin if you have time... 

https://github.com/ebullient/obsidian-task-collector/releases/tag/1.0.0

Will make sure your config is adapted correctly. ;)

**Comment by ebullient** [comment]
ok. regarding these: 

        "rightClickCycle": true,
        "rightClickCreate": true,
        "rightClickAppend": true,

I have a setting for the cycle, but it isn't implemented yet. Do you really want a menu item? or is a command you can bind to a hot key better? 

Oh.. actually. I have a question. Should I drop all/most right click menu items? You can use the commander plugin to make any command a menu item.. so feels like just providing commands would allow more customization of menus... WDYT?

The right click append is how you append the configured strings to the text without making it a task, right? (is that the correct description of behavior?) .. does that also apply the remove text regex? From the modal, you can choose a status for a regular line item (in base task collector), and it will make it a marked task and apply append/remove settings... 

Is "incompleteTaskValuesRow2" used to render the marks in a significant way? Marks will show up in a consistent (sorted) order complete in the top row, incomplete in the second row/group (those will wrap). Is that enough?

**Comment by RobColes** [comment]
Think I'm doing something dim with the 1.0.0 tag version. I've pulled down the .js, css and manifest, but I can't get into settings.
TaskColl1 screenshot is on entry to the vault with new task collector enabled
TaskColl2 is on clicking on the settings gear against the plug in
TaskColl3 is clicking on the X to try to close the setting window.

![TaskColl3](https://user-images.githubusercontent.com/1765065/211808274-96e19659-9132-4d9d-9243-864b6d7df12e.png)
![TaskColl2](https://user-images.githubusercontent.com/1765065/211808282-7ce758a0-be95-481a-b2be-dd493d113daf.png)
![TaskColl1](https://user-images.githubusercontent.com/1765065/211808283-62de00f7-9fb6-4271-9bab-f8c2ec9b1f19.png)

**Comment by wenlzhang** [comment]
> Hotkeys (and context menu additions) are in the mix for my revisions, so .. we'll see what you think. ;)

Looking forward to that!

> This is an interesting use case. For commands (hot-keyable) in edit mode, a flag could allow text to be appended to a line that isn't a task (turn off any "find the task" matching capability and just append.. ?). Do the strings that you use in this case also overlap with task types?

Yes. It would just append text and does not check the task type. The strings used to append text is are the same moment strings as those for tasks.

> Should I drop all/most right click menu items? You can use the commander plugin to make any command a menu item.. so feels like just providing commands would allow more customization of menus... WDYT?

I think I would rather keep those menu items, just in case some users would not like to install an additional plugin for such a purpose.

> The right click append is how you append the configured strings to the text without making it a task, right? (is that the correct description of behavior?) .. does that also apply the remove text regex?

Yes. The appending feature does not make the line a task; it just appends the text. It does not involve removing the text because I did not find a use case for “resetting text” yet…

> Is "incompleteTaskValuesRow2" used to render the marks in a significant way? Marks will show up in a consistent (sorted) order complete in the top row, incomplete in the second row/group (those will wrap). Is that enough?

In the current implementation, there are three rows in the mark modal. The first row includes `[x], [-]`, etc.; the second includes additional marks apart from those included in the first row, and text would be appended; for the third row, no text would be appended.

"incompleteTaskValuesRow2" is for the marks in the third row of the mark modal.

**Comment by wenlzhang** [comment]
> Most of my use of the different statuses is within a daily note, so it may be that I think I've finished a ticket, but the user comes back with a further issue, or occasionally a cancelled call gets re-instated.

I see this point. I have this use case too.

> My use case doesn't automatically append anything, I am really just looking at changing the status so multiple appends aren't an issue for me.

I see. I would improve this point later. The first step would be to allow marking completed/cancelled tasks as a different status. The second step would be to give a reminder if the task is already marked with the same mark.

**Comment by RobColes** [comment]
>>Oh.. actually. I have a question. Should I drop all/most right click menu items? You can use the commander plugin to make any command a menu item.. so feels like just providing commands would allow more customization of menus... WDYT?

I'd vote to leave them too - the way you've done it, they can be switched off if not wanted. I don't use commander either.

**Comment by ebullient** [comment]
Will respin to fix the log issue.

> In the current implementation, there are three rows in the mark modal. The first row includes [x], [-], etc.; the second includes additional marks apart from those included in the first row, and text would be appended; for the third row, no text would be appended.

This is handled group by group now. So the modal does separate "complete" items (top row) from other items (everything else), but the behavior for each mark (complete or not) depends on the group it is in (irrespective of where it appears in the modal).

appendTextFormatCreation -- Making sure I understand. Task Collector will already convert a list item to a marked task (and perform configured append/remove). This allows _any_ line of text to be .. made a task? Or this is your generic append case (nothing to do with tasks).

Sorry if you feel like you're explaining things on repeat. Just making sure I understand.

**Comment by wenlzhang** [comment]
> appendTextFormatCreation -- Making sure I understand. Task Collector will already convert a list item to a marked task (and perform configured append/remove).

This is to convert a list item as an open task and append text. Can Task Collector mark a list item as an open task and append text? I haven't used this in Task Collector before. Another thing with this is that one can assign hotkeys for this command.

> This allows any line of text to be .. made a task?

This is actually a planned feature.

> Sorry if you feel like you're explaining things on repeat. Just making sure I understand.

No worries at all. Glad that we can discuss such things. :-)

**Comment by ebullient** [comment]
> > appendTextFormatCreation -- Making sure I understand. Task Collector will already convert a list item to a marked task (and perform configured append/remove).
> 
> This is to convert a list item as an open task and append text. Can Task Collector mark a list item as an open task and append text? I haven't used this in Task Collector before. Another thing with this is that one can assign hotkeys for this command.

I believe this should have been possible before (use the modal, select space..) if you had "appendRemoveAll" enabled, but the new version is a lot better in this regard.
 
> > This allows any line of text to be .. made a task?
> 
> This is actually a planned feature.

Ok. Task Collector should be able to do this now (w/ a group that includes a space (` `) as the marker. I will double check the "any old line" part. You could already create a task from a list item using the modal and typing a character: `- some stuff` -> `- [x] some stuff (date)`.

I'll add some tests for this.

> > Sorry if you feel like you're explaining things on repeat. Just making sure I understand.
> 
> No worries at all. Glad that we can discuss such things. :-)

**Comment by wenlzhang** [comment]
> The first step would be to allow marking completed/cancelled tasks as a different status. The second step would be to give a reminder if the task is already marked with the same mark.

@RobColes I released a new version that addresses these. Maybe you could give it a try?
https://github.com/wenlzhang/obsidian-task-marker/releases/tag/0.3.1

**Comment by ebullient** [comment]
I made a BRAT release for 1.0.0.. will just keep moving forward from there. Manual install is too fussy.
I have the config for the task cycle, but not the command. Will add that shortly.

If you make a group called `text`, it should do the strict text append... there are probably some rough edges around command names and things, but the broad strokes should work.

**Comment by wenlzhang** [comment]
> If you make a group called `text`, it should do the strict text append... there are probably some rough edges around command names and things, but the broad strokes should work.

Haven't tested it intensively yet, but I like the grouping feature.

**Comment by wenlzhang** [comment]
> If you make a group called `text`, it should do the strict text append

@ebullient Would it be possible to support appending different types of text in different use cases, e.g. `📝 2023-01-08T12:00`, `❎ 2023-01-08T12:00` and `✅ 2023-01-08T12:00`. They correspond with different groups of tasks with different dates to append.

**Comment by ebullient** [comment]
> > If you make a group called `text`, it should do the strict text append
> 
> @ebullient Would it be possible to support appending different types of text in different use cases, e.g. `📝 2023-01-08T12:00`, `❎ 2023-01-08T12:00` and `✅ 2023-01-08T12:00`. They correspond with different groups of tasks with different dates to append.

I wondered about that. ;) 

What do you think.. different groups (starting with text)? Or, put the group selector back (so you could put multiple emoji in the box), and select between those? It would be a bit different, as tasks rely on CSS to render the emoji based on easy-to-type characters, and that isn't as doable with arbitrary text.

Templater is ideal for this, as you can bind arbitrary text strings to hotkeys (it also uses moment.js format for dates)

**Comment by wenlzhang** [comment]
> different groups (starting with text)?

I think this option allows more customization because one may append different types of text, e.g. one with Dataview inline field but another one without.

> It would be a bit different, as tasks rely on CSS to render the emoji based on easy-to-type characters, and that isn't as doable with arbitrary text.

I guess this refers to how to render the task icon like `[x]` in the selection modal? For the text appending case, one option could be allowing the user to configure the text to show in the selection modal (otherwise, the first character of the text to append would be used). For example, I might use `📝`, `❎` and `✅`, respectively.

> Templater is ideal for this, as you can bind arbitrary text strings to hotkeys (it also uses moment.js format for dates)

I am not certain if one can use Templater to achieve such a purpose in a simple way like Task Collector for tasks, where one can see and configure everything in the settings without relying on other scripts or notes within the vault.

**Comment by ebullient** [comment]
> I guess this refers to how to render the task icon like [x] in the selection modal?

Yes. And then how you can select them (what the task looks like).

> allowing the user to configure the text to show in the selection modal (otherwise, the first character of the text to append would be used). 

This could work. I think it means a different kind of config group altogether.

> I am not certain if one can use Templater to achieve such a purpose in a simple way like Task Collector for tasks, where one can see and configure everything in the settings without relying on other scripts or notes within the vault.

This part is true.

**Comment by wenlzhang** [comment]
> I think it means a different kind of config group altogether.

Yes. Maybe there can also be a separate section for this. Something like "Text Groups"?

**Comment by ebullient** [comment]
Let me chew on it for a few days (while I make sure everything else is good)

**Comment by wenlzhang** [comment]
> 1. If I cycle through the statuses with the cursor past the last of the text, the cursor position is left alone. If I am e.g. in the middle of a word, as soon as I cycle via hotkey, the cursor goes to the start of the line and stays there.  I would prefer the cursor position to be left alone if possible.

@RobColes The cursor position issue should have been fixed with version 0.4.2. Could you try and see if it resolves the issue for you?

**Comment by RobColes** [comment]
@wenlzhang Thank you - I have run a few tests on the cursor positioning and it is all exactly as I would expect. I'll keep an eye on it but it looks great now.

Changing the status of a closed task directly also works for me now, so the ticket is completed.

Many thanks for fixing this!
